### PR TITLE
Cleaning up some unit test warnings in console

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description
We had some unit tests that were producing warnings in the console, which makes it difficult to see which tests are failing and why (especially in non-color consoles, like sometimes here on github).

## Fixed
- Removing leftover `screen.debug()`
- Adding missing keys to validation messages (which React complains about). I struggled to find something unique, so serializing all the arguments should be unique enough. 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- ~~[ ] Manual testing done (required)~~
- ~~[ ] Relevant automated test added (if you find this hard, leave it and we'll help out)~~
- [x] All tests run green

## Documentation
- ~~[ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)~~

### CODE REVIEW

File: src/altinn-app-frontend/src/utils/render.tsx [hunk 1]
```diff
@@ -61,16 +61,24 @@ export function renderValidationMessages(
...
   variant: 'error' | 'warning' | 'info' | 'success',
 ) {
+  const key = JSON.stringify({ messages, id, variant });
+
   if (variant !== 'error') {
     return (
-      <SoftValidations variant={variant}>
+      <SoftValidations
+        variant={variant}
+        key={key}
```
Comment (haakemon) [inline src/altinn-app-frontend/src/utils/render.tsx:70]: This seems odd to me - there is no looping over values here, so why should a key be nescessary?

I think the underlying problem here is the `renderValidationMessagesForComponent` code that returns an array of items directly, and that is returned, and used as a component (rendered). 

Shouldnt this be fixed instead by actually looping over the items and add a key there?


On a sidenote, shouldnt `id` be unique?

Comment (olemartinorg) [reply src/altinn-app-frontend/src/utils/render.tsx:70]: Haha, yeah, you're right - you exposed my laziness.. :wink: I'll look into rewriting this to something better.

As for the `id` not being unique - yeah, I dunno.. I printed out all the `id` arguments given during the full test suite, and there were a bunch of duplicate IDs where `undefined` had been passed in (and just converted to a string at some point). It seems fickle and buggy to me, so I agree a proper cleanup is in order.

Comment (haakemon) [reply src/altinn-app-frontend/src/utils/render.tsx:70]: I mean this "fixes" the problem, and definitely improves the output from tests - so maybe its worth merging this as-is now, and do a proper cleanup later. I assume that task is a bit bigger, and will require more testing. You have my approval at least 🙂

**Review by haakemon — APPROVED** [review]

File: src/altinn-app-frontend/src/utils/render.tsx (hunk 1 above)
Comment (olemartinorg) [reply src/altinn-app-frontend/src/utils/render.tsx:70]: Thanks! I'll look into it, but won't spend a week on it.. :wink:

**Pull request merged**